fix(docs-infra): fix `autoLinkCode` Dgeni post-processor

gkalpak opened the pull request:
A couple of fixes related to auto-linking `<code>` elements in the docs.
(See individual commits for details.)

mary-poppins commented:
You can preview 38128e3 at https://pr41709-38128e3.ngbuilds.io/.

petebacondarwin commented on aio/tools/transforms/angular-base-package/post-processors/auto-link-code.js:
```diff
@@ -46,6 +46,10 @@ module.exports = function autoLinkCode(getDocFromAlias) {
         }
 
         visit(node, 'text', (node, ancestors) => {
+          if (node.addedByAutoLinkCode) {
+            return;
+          }
+
```
I think we can do this without adding properties to the nodes...
In the case where we call `getNodes()` I think we can indicate to the `visit()` that we do not want to recurse into the text further. See https://github.com/syntax-tree/unist-util-visit-parents#next--visitornode-ancestors. I think we could do something like:


```ts
} else {
            // Parse the text for words that we can convert to links
            const nodes = getNodes(node, file);
            // Replace the text node with the links and leftover text nodes
            Array.prototype.splice.apply(parent.children, [index, 1].concat(nodes));
            return [SKIP, index+1];
          }
```

gkalpak replied:
I added a fixup commit. PTAL

petebacondarwin reviewed (changes requested):
Can you look into skipping the whole generated tree instead of patching the text node?

gkalpak committed:
fix(docs-infra): apply custom `autoLinkCode` filters to single-word `<code>` elements

Previously, the `autoLinkCode` Dgeni post-processor would not apply the
custom filters when matching the whole contents of a `<code>` element.
This meant that custom filters would not be applied to single-word
`<code>` elements.

You can see occurrences of this issue in the following sections of the
"Reactive forms" guide:
- [Creating nested form groups][1]
  (look for `street, city, state, and zip controls`)
- [Using the FormBuilder service to generate controls][2]
  (look for `group method`)

This commit fixes this by also applying the custom filters when
processing the whole contents of a `<code>` element.

This commit also updates the `filterPipes` custom filter to allow
matching a pipe's name in a single-word `<code>` element (where there is
no preceeding `|` character).

[1]: https://v10.angular.io/guide/reactive-forms#creating-nested-form-groups
[2]: https://v10.angular.io/guide/reactive-forms#using-the-formbuilder-service-to-generate-controls

gkalpak committed:
fix(docs-infra): do not process generated nodes for auto-linking

While generating the docs, when a `<code>` element is inspected for
auto-linking, the `autoLinkCode` Dgeni post-processor will break its
contents up into words and generate text nodes for those words that
should not be auto-linked.

Previously, our text node visitor would visit these generated text nodes
and try to auto-link them too. As a result, it would unnecessarily
process nodes that had already been checked (and could potentially
generate links that would otherwise be ignored).

You can see an occurrence of this issue in the
[Create the product list][1] section of the
"Getting started with Angular" tutorial (look for `<a>`).

This commit fixes this by ensuring the visitor will skip the current
node and any nodes generated by `autoLinkCode`.

[1]: https://v11.angular.io/start#create-the-product-list

gkalpak committed:
fixup! fix(docs-infra): do not process generated nodes for auto-linking

mary-poppins commented:
You can preview fe37001 at https://pr41709-fe37001.ngbuilds.io/.

petebacondarwin reviewed (approved):
LGTM! Nice fixes

mary-poppins commented:
You can preview 7ca4b6f at https://pr41709-7ca4b6f.ngbuilds.io/.

jessicajaniuk commented:
@gkalpak This PR has merge conflicts with `11.2.x`. It's been merged into `master` and `12.0.x`. Can you create a patch CL targeting `11.2.x`?

gkalpak commented:
For reference, here is the PR for patch: #41854
